### ISSUE TITLE
Connections handling

### ISSUE THREAD

**Issue by larev** [grpc/grpc-java#1241]
Hello! Is it possible to handle connections(their statuses) on the server side?

**Comment by ejona86** [comment]
It's not clear to me what you are wanting to do. What is your use-case?

**Comment by larev** [comment]
I want to do something like JMS only using the grpc. The problem is that I can't clear the listeners without the status of client connections on the server side.

**Comment by ejona86** [comment]
For handling listeners I'd suggest you use a stream. Each event can then be its own response on the stream. If the client goes away then the stream will be closed and you can reclaim resources. If the TCP connection goes down the client will be informed so that it can create a new RPC to listen.

**Comment by larev** [comment]
Ok, help me please...
Pseudo code:
```java
public class DeviceServiceImpl extends DeviceServiceGrps.DeviceService{
    public void listenDevice(DeviceId id, StreamObserver<DeviceEvent> response){
        Iterator<DeviceEvent> events=SuperDeviceEvents.getBlockingIterator(id);
        while(events.hasNext())
        {
            response.onNext(events.next())
        }
    }
}
```
How can I abort the iteration in this situation

**Comment by ejona86** [comment]
The blocking iterator doesn't currently provide a way to abort. If you use the async interface, then calling `StreamObserver.onError()` will cancel the stream.

0.10 will have context integration, so you could use the blocking interface like:
```java
try (Context.current().withCancellation().attachAsCloseable()) {
    Iterator<DeviceEvent> events = SuperDeviceEvents.getBlockingIterator(id);
    while(events.hasNext())
    {
        response.onNext(events.next())
    }
}
```

And then simply leaving the try block will cancel the RPC.

**Comment by ejona86** [comment]
0.12.0 is now released (it was originally going to be named 0.10).
